cmd/appliance/upgrade/complete.go

```diff
@@ -47,7 +47,7 @@ type upgradeCompleteOptions struct {
 	actualHostname    string
 	defaultFilter     map[string]map[string]string
 	ciMode            bool
-	batchSize         int
+	maxUnavailable    int
```

Comment:
They don't have to be exported. The flag interacts with the variable by taking its pointer as an argument, thus manipulating the value directly. You can see the flag definition further down in the code here:

Reply:
I saw that piece I think I am just misunderstanding how the use uses this. The presence of what looks to be help info in that snippet you posted is what makes me think it isnt actually exported but is still part of our API.

Said another way can a user provide --max-unavailable as an argument to the tool? If the answer to that is yes then max-unavailable is part of the api even if this exact struct isnt used.

I am trying to understand if the user will set the variable. Seems like this has to be yes.

I am also trying to understand the second question about 1 max unavailable flag vs 1 flag per type. so one for gateways and one for logforwarders.

Reply:
Well, it's definitely part of the API, and the help text is how you would set it to provide the help for the user in the help section. The struct is not itself part of the API. You can view it more as a value store to save the values set by the user throughout the command rather than having to query for which value was set when we need it. It keeps the codebase a bit more organized.

As for why we don't have 1 flag per type, it's not really something that's been up on the table. The main concern has been the gateways, but the logforwarders still need to abide by the same rules, so we need to address it anyway. We could probably have 1 flag per type, but, in my view, that would needlessly overcomplicate things.